tests/std/tests/P0660R10_jthread_and_cv_any/test.cpp

```diff
@@ -0,0 +1,244 @@
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include <assert.h>
+#include <condition_variable>
+#include <mutex>
+#include <stdio.h>
+#include <thread>
+#include <type_traits>
+#include <utility>
+
+#define STATIC_ASSERT(...) static_assert(__VA_ARGS__, #__VA_ARGS__)
+
+#ifdef __clang__
+#pragma clang diagnostic ignored "-Wself-move"
+#endif // __clang__
+
+using namespace std;
+
+STATIC_ASSERT(is_same_v<thread::id, jthread::id>);
+STATIC_ASSERT(is_same_v<thread::native_handle_type, jthread::native_handle_type>);
+
+int main() {
+    // dtor tested in lots of places here so no explicit tests for it
+
+    { // default ctor
+        jthread default_constructed;
+        assert(default_constructed.get_id() == thread::id{});
+        assert(!default_constructed.get_stop_source().stop_possible());
+    }
+
+    { // initializing ctor, traditional functor
+        jthread worker{[] {}};
+        assert(worker.get_id() != thread::id{});
+        assert(worker.joinable());
+        assert(worker.get_stop_source().stop_possible());
+    }
+
+    { // also make sure that we don't delegate to std::thread's constructor which would try to move assign over the
+      // std::thread inside jthread rather than passing it to the functor
+        jthread worker{[](thread t) { t.join(); }, thread{[] {}}};
+        assert(worker.get_id() != thread::id{});
+        assert(worker.joinable());
+        assert(worker.get_stop_source().stop_possible());
+    }
+
+    { // initializing ctor, token functor
+        bool called = false;
+        struct overload_detector {
+            bool* p_called;
+            void operator()(stop_token, int i) const {
+                assert(i == 1729);
+                *p_called = true;
+            }
+            void operator()(int) const {
+                assert(false);
+            }
+        };
+
+        {
+            jthread worker{overload_detector{&called}, 1729};
+            (void) worker;
+        }
+
+        assert(called);
+    }
+
+    { // move ctor
+        jthread worker{[] {}};
+        auto worker_source = worker.get_stop_source();
+        {
+            jthread moved{move(worker)};
+            assert(moved.get_stop_source() == worker_source);
+            assert(moved.joinable());
+            assert(worker.get_stop_source() != worker_source);
+            assert(!worker.joinable());
+        }
+    }
+
+    { // move assign
+        jthread worker_a{[] {}};
+        auto source_a = worker_a.get_stop_source();
+        jthread worker_b{[] {}};
+        auto id_b     = worker_b.get_id();
+        auto source_b = worker_b.get_stop_source();
+        worker_a      = move(worker_b);
+        assert(source_a.stop_requested());
+        assert(id_b == worker_a.get_id());
+        assert(!source_b.stop_requested());
+        assert(worker_a.get_stop_source() == source_b);
+        assert(!worker_b.joinable());
+    }
+
+    { // self move assign, currently specified to try to cancel and join

```

Comment:
Standard citation would be helpful here (otherwise "currently" will become confusing).